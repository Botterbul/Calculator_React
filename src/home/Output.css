.Output {
    width: 100%;
    text-align: right;
    font-size: 2.5rem;
    padding-right: 50px;
    background-color: #1a0000;
    color: white;
    width: 100%;
}