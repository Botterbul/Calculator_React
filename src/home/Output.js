import React from 'react';
import './Output.css';

class Output extends React.Component {
  static propTypes = {
    value: PropTypes.string,
  };

  render() {
    return (
      <div className="Output">
        <div>{this.props.value}</div>
      </div>
    );
  }
}

export default Output;