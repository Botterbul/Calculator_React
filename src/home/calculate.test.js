const { isNumber } = require('./isNumber');
const { operate } = require('./operate');
import React from 'react';
import renderer from 'react-test-renderer';
import Output from './Output';

// Test Cases for the isNumber Method
test('Should be seen as number', () => {
    const answer = isNumber("0");
    expect(answer).toBe(true);
});

test('Should not seen as number', () => {
    const answer = isNumber("A");
    expect(answer).toBe(false);
});

//======================================

// Test Cases for the operate Method
test('Test Add Method with only zeros', () => {
    const answer = operate("0", "0", "+");
    expect(answer).toBe("0");
});

test('Test Add Method', () => {
    const answer = operate("7", "0", "+");
    expect(answer).toBe("7");
});

test('Test Subtract Method', () => {
    const answer = operate("7", "2", "-");
    expect(answer).toBe("5");
});

test('Test Multiply Method', () => {
    const answer = operate("7", "2", "x");
    expect(answer).toBe("14");
});

test('Test Subtract Method', () => {
    const answer = operate("7", "2", "x");
    expect(answer).toBe("14");
});

test('Test Divide Method', () => {
    const answer = operate("8", "4", "รท");
    expect(answer).toBe("2");
});

//Snapshot testing
describe('Output Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Output value="0"/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

