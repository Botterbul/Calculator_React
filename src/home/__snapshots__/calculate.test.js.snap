// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Component matches the snapshot 1`] = `
<div
  className="Output"
>
  <div>
    0
  </div>
</div>
`;

exports[`Panel Component matches the snapshot 1`] = `
<div
  className="Output"
>
  <div>
    0
  </div>
</div>
`;
